src/main/scala/io/iohk/ethereum/utils/VersionInfo.scala

```diff
@@ -0,0 +1,48 @@
+package io.iohk.ethereum.utils
+
+object VersionInfo {
+
+  /** Produce a node name that get sent in `WireProtocol.Hello.clientId` according to Ethereum conventions.
+    *
+    * Check out examples on https://etcnodes.org
+    *
+    * e.g.
+    * - mantis/v3.0-cd5ae33/linux-amd64/ubuntu-openjdk64bitservervm-java-11.0.9
+    * - besu/v20.10.0/linux-x86_64/oracle_openjdk-java-11
+    * - coregeth/v1.11.8-stable-305b5089/linux-amd64/go1.14.4
+    *
+    * Apparently ethstats expects either 4 parts or 5:
+    * - client/version/os/compiler
+    * - client/identity/version/os/compiler
+    */
+  def nodeName(maybeIdentity: Option[String] = None): String = {
+    val app = {
+      val name = BuildInfo.name
+      val id = maybeIdentity.map("/" + _).getOrElse("") //
+      s"$name$id"
+    }
+    val version = {
+      val version = BuildInfo.version
+      val commit = BuildInfo.gitHeadCommit.map("-" + _.take(7)).getOrElse("")
```

Comment:
Minor: From the example `coregeth/v1.11.8-stable-305b5089/linux-amd64/go1.14.4`, shouldn't it be `take(8)`?

Though I'm not sure why they do something different than GH... and also that change probably has no impact so we can diverge on that

Reply:
I used 7 to match github. I think the version is largely up to us, I thought it's useful since you can test something without changing the version, and as long as we can use the commit hash in Github to find the exact code version it looks useful.